File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    
    heapq.heapify(scoville)

    while scoville:
        first = heapq.heappop(scoville)
        if first >= K:
            break
        if len(scoville) == 0 and first < K:
            answer = -1
            break
        second = heapq.heappop(scoville)
        mixed = first + (second * 2)
        heapq.heappush(scoville, mixed)
        answer += 1
    
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        // scoville을 우선순위큐로 만들기
        Queue<Integer> pq = new PriorityQueue<>();
        for (int i = 0; i < scoville.length; i++) {
            pq.offer(scoville[i]);
        }
        
        while (!pq.isEmpty()) {
            int first = pq.poll();
            
            if (first >= K) break;
            
            if (pq.isEmpty()) {
                answer = -1;
                break;
            } 
            
            int second = pq.poll();
            pq.offer(first + (second * 2));
            answer++;
        }
        
        return answer;
    }
}